### PR TITLE
Fix Emitter to handle comments between a mapping key and a mapping/sequence value

### PR DESCRIPTION
**Pull request by YihaoOct**
This PR fixes https://github.com/aaubry/YamlDotNet/issues/812

TL;DR the Emitter doesn't handle comment events that are emitted between a mapping key and a mapping value of the mapping or sequence type. So parsing and regenerate the following yaml will fail for example:

```yaml
hello:
  # comment
  - world
```

becomes
```yaml
hello
# this is a comment
:
- world
```

This PR moves the writing of the `:` symbol to be inside the block mapping key emission. Previously it was at the beginning of the block mapping value emission. If the mapping key is prefixed with a `?` (explicit form), however, the `:` is still kept at the mapping value to preserve the existing behaviour, since in the explicit form, the comment does not break YAML syntax. That is,
```yaml
? hello
# this is a comment
:
- world
```
is valid.

### CODE REVIEW

**Comment by aaubry** [comment]
This feature has been released in [version 13.1.1](https://github.com/aaubry/YamlDotNet/releases/tag/v13.1.1).
